instrumentation/net/http/otelhttp/handler.go

```diff
@@ -219,21 +219,3 @@ func (h *middleware) metricAttributesFromRequest(r *http.Request) []attribute.Ke
 	}
 	return attributeForRequest
 }
-
-// WithRouteTag annotates spans and metrics with the provided route name
-// with HTTP route attribute.
-//
-// Deprecated: spans are automatically annotated with the route attribute.
-// To annotate metrics, use the [WithMetricAttributesFn] option.
```

Comment:
Non-blocking: If we are able to annotate spans automatically, why don't we annotate metrics automatically as well?

From https://opentelemetry.io/docs/specs/semconv/http/http-metrics/#metric-httpserverrequestduration for http.route:

> Conditionally Required If and only if it’s available

Reply:
No idea. I expect it should be possible.

Reply:
If you would like to open an issue to track this I would support that.

I'm going to resolve the comment as it doesn't seem related to the removal being done in the PR.